crates/bevy_ecs/src/schedule/label.rs

```diff
@@ -25,3 +25,75 @@ define_label!(
     /// Strongly-typed identifier for a [`RunCriteriaLabel`].
     RunCriteriaLabelId,
 );
+
+/// Defines one or more local, unique types implementing [`StageLabel`].
+#[macro_export]
+macro_rules! stage_label {
```

Comment:
Why add these macros when there's already a derive macro?
And why `allow(non_camel_case_types)`? That's kinda cursed.

Reply:
This macro is a lot more terse than the derive macro if you want to declare multiple label types. I allow snake case because I want it to feel like something of a replacement for string labels, and PascalCaseTypes stand out way too much. If it's the only PascalCase identifier on a line, then your attention will get drawn to the label even though it's arguably the least important information there.

Plus, I think allowing "wild west" names is okay, since they're all *guaranteed* to be private. I expect most of the uses will just be for unit tests, like I've shown here.

Reply:
There's also precedent for generating private types with non-standard naming styles -- for one example see [syn::custom_keyword](https://docs.rs/syn/latest/syn/macro.custom_keyword.html).

Reply:
I'd prefer to use standard type naming conventions here; I think it reduces surprise when reading the code. On board with the easy generation of these types for tests though.